### PR TITLE
Add route to create a new analysis

### PR DESCRIPTION
**Pull request by dmfalke**
This is needed for links to create a new analysis that will appear on the ClinEpiDB homepage and menu.

### CODE REVIEW

File: src/lib/workspace/NewAnalysis.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        derivedVariables: [],
+        visualizations: [],
+        computations: [],
+        variableUISettings: {},
+      });
+      const newLocation = {
+        ...history.location,
+        pathname: `./${id}`,
+      };
+      history.push(newLocation);
```
Comment (jtlong3rd) [inline src/lib/workspace/NewAnalysis.tsx:29]: I think we should "cancel" this redirect if the user navigates away from the "new analysis" route before the analysis creation has finished. What do you think?

**Review by jtlong3rd — COMMENTED** [review]
Looks good! I had one inline comment about the current UX.

File: src/lib/workspace/NewAnalysis.tsx (hunk 1 above)
Comment (dmfalke) [reply src/lib/workspace/NewAnalysis.tsx:29]: Yep, I agree with that. Do we have any utilities that would help in this case?

Comment (dmfalke) [reply src/lib/workspace/NewAnalysis.tsx:29]: I forgot about this utility class: https://github.com/VEuPathDB/WDKClient/blob/af2099baf3048f65d759023c2c7837d1b359d70b/Client/src/Utils/Task.ts. I might try it out

Comment (jtlong3rd) [reply src/lib/workspace/NewAnalysis.tsx:29]: Neat! If that doesn't pan out, I suppose you could always... 

1. Do the "usual" trick of adding a `cancelled` or `ignore` variable to the effect

OR

2. Use `usePromise` to create the analysis and conditionally render a `Redirect`

Comment (dmfalke) [reply src/lib/workspace/NewAnalysis.tsx:29]: I do think (1) is the easiest to read.

This is what it looks like with `Task`:

```typescript
  useEffect(
    () =>
      Task.fromPromise(
        analysisStore.createAnalysis({
          name: 'Unnamed Analysis',
          studyId,
          filters: [],
          starredVariables: [],
          derivedVariables: [],
          visualizations: [],
          computations: [],
          variableUISettings: {},
        })
      ).run(({ id }) => {
        const newLocation = {
          ...history.location,
          pathname: `./${id}`,
        };
        history.push(newLocation);
      }),
    [analysisStore, history, studyId]
  );
```

Comment (jtlong3rd) [reply src/lib/workspace/NewAnalysis.tsx:29]: Let's go with (1) then!

Comment (dmfalke) [reply src/lib/workspace/NewAnalysis.tsx:29]: For the sake of comparison
```typescript
  useEffect(() => {
    let cancelled = false; 
    analysisStore.createAnalysis({
      name: 'Unnamed Analysis',
      studyId,
      filters: [],
      starredVariables: [],
      derivedVariables: [],
      visualizations: [],
      computations: [],
      variableUISettings: {},
    }).then(({ id }) => {
      if (!cancelled) {
        const newLocation = {
          ...history.location,
          pathname: `./${id}`,
        };
        history.push(newLocation);
      }
    });
    return () => { cancelled = true };
  }, [analysisStore, history, studyId]);
```

Comment (dmfalke) [reply src/lib/workspace/NewAnalysis.tsx:29]: The thing I like about the version with `Task` is that there are fewer variables floating around, thus less potential for something to break in the future (not likely here, but in general).

Comment (jtlong3rd) [reply src/lib/workspace/NewAnalysis.tsx:29]: Using Task is fine by me! (And I'll definitely have to keep in mind that we have this utility at our disposal.)

Comment (dmfalke) [reply src/lib/workspace/NewAnalysis.tsx]: Okay, I pushed a version using `Task` :shrug:

**Review by jtlong3rd — APPROVED** [review]

**Pull request merged**